582864 upgrade to spring 6 and osgi 7 and several unit tests to junit 4

mxcross opened the pull request:
upgrade to spring 6 and osgi 7 and several unit tests to junit 4

mxcross committed:
582864 upgrade to spring 6 and osgi 7 and several unit tests to junit 4

mxcross committed:
582864 update osgi.framework and osgi.service.cm versions

mxcross committed:
582864 remove SecurityUtils

kennymacleod commented on pom.xml:
```diff
@@ -23,29 +23,31 @@
 
         <!-- compiled against osgi Revision -->
-        <osgi.ver>5.0.0</osgi.ver>
+        <osgi.ver>7.0.0</osgi.ver>
```
why not 8.0.0?

mxcross replied:
7.0.0 is the minimum version needed to make it compatible with the felix and equinox updates

wilx commented:
How does this differ from #11, #12, #13? And there is also #10.

wilx commented:
I am not a boss here but please, can you guys combine your work and get this new version rolling?

mxcross commented:
> How does this differ from #11, #12, #13? And there is also #10.

#11, #12, and #13 are all spring 6.0, not 6.1.

#10 updates the spring library but does not have the changes to make the code or tests work. It actually is the branch I started with to make the changes, but it was done on a non-public github.

mxcross committed:
582864 incorporate changes from PR #13

wilx commented:
I believe the changes from #16 will be necessary as well for this to work.

mxcross committed:
582864 incorporate changes from PR #16

mxcross committed:
582864 upgrade to spring 6.1.6_1 which fixes the issue with
javax.xml.parsers exceptions

mxcross committed:
582864 update calls to new Long to Long.valueOf

mxcross committed:
582864 add spring-boot-maven-plugin to resolve exceptions when spring
aot is accessed. Execution of process-aot is how the spring class
generation occurs, it is no longer in the experimental spring-native
library

mxcross committed:
Revert "582864 add spring-boot-maven-plugin to resolve exceptions when spring aot is accessed. Execution of process-aot is how the spring class generation occurs, it is no longer in the experimental spring-native library"

This reverts commit cf3eba6f2772a36da1e9951c6d53b6588a3b81ed.

mxcross commented:
> I believe the changes from #16 will be necessary as well for this to work.

Thank you for the update! I did include it, but I'm still getting a failure on the continuous-integration step. It seems it's not using the update to the Jenkinsfile because I'm not considered trusted.

wilx commented:
> > I believe the changes from #16 will be necessary as well for this to work.
> 
> Thank you for the update! I did include it, but I'm still getting a failure on the continuous-integration step. It seems it's not using the update to the Jenkinsfile because I'm not considered trusted.

Neither am I. I am not sure what to do about it. @olaf-otto?

wilx commented:
Are these necessary in compile scope? It seems to me they should be at best test dependencies:
https://github.com/mxcross/gemini.blueprint/blob/acac7204e1bb2b5069dde34459b7f896ab1fd442/pom.xml#L358-L384

mxcross committed:
582864 upgrade to osgi 8.0.0

mxcross commented:
> Are these necessary in compile scope? It seems to me they should be at best test dependencies: https://github.com/mxcross/gemini.blueprint/blob/acac7204e1bb2b5069dde34459b7f896ab1fd442/pom.xml#L358-L384

those dependencies are for logging as the comment suggests in the pom.

wilx commented:
No part of this project is using SLF4J or Logback. The commons-logging `Log` class can be provided either by `spring-jcl` or by `commons-logging`.

mxcross commented:
> No part of this project is using SLF4J or Logback. The commons-logging `Log` class can be provided either by `spring-jcl` or by `commons-logging`.

although no references are in the java code, there are references in the bundle configurations. I'd rather not remove dependencies I didn't add that would impact logging.

olaf-otto commented:
> > > I believe the changes from #16 will be necessary as well for this to work.
> > 
> > 
> > Thank you for the update! I did include it, but I'm still getting a failure on the continuous-integration step. It seems it's not using the update to the Jenkinsfile because I'm not considered trusted.
> 
> Neither am I. I am not sure what to do about it. @olaf-otto?

Hi! I have access to Jenkins, not sure what the message "‘Jenkinsfile’ has been modified in an untrusted revision" will yield, but it looks like a warning. The actual issue is:

````
WorkflowScript: 47: Duplicate stage name: "Build with Equinox profile" @ line 47, column 3.
     stages {
````

@mxcross I will try and apply your changes from the Jenkinsfile on main directly.

olaf-otto commented:
@mxcross I've applied the changes to main - can you please modify the PR accordingly? Thanks!

mxcross committed:
Merge branch 'eclipse-gemini:main' into main

mxcross commented:
> @mxcross I've applied the changes to main - can you please modify the PR accordingly? Thanks!

Updated. Thank you!

mxcross committed:
582864 update version of integration-tests to the same as the rest of
the project

mxcross committed:
582864 update integration-tests so they compile

mxcross committed:
582864 continue upgrading tests from junit 3 to 4

mxcross committed:
582864 convert integration tests to junit 4 (still not functional but
more errors we can work off of since junit 3 support has been dropped by
spring)

mxcross commented:
@olaf-otto I'm finding that even on the main branch that the integration-tests folder doesn't compile nor does it pass if you go through the trouble of getting them to compile. Is there some plan around these since they're gating any PRs from the repo?

olaf-otto commented:
> @olaf-otto I'm finding that even on the main branch that the integration-tests folder doesn't compile nor does it pass if you go through the trouble of getting them to compile. Is there some plan around these since they're gating any PRs from the repo?

Oh, I was unaware of that situation. I shall take a look tonight, it should compile and pass indeed. Thanks for pointing that out!

olaf-otto commented:
Hi @mxcross I have fixed the integration tests on the main branch. There, I reverted to JDK 8 since the main branch is still on that language level. I have also made sure that clean integration testing cycles are run for each tested OSGi container.

Thus, the expectation is that the integration test shall also pass after the upgrade. Migrating the integration tests accordingly might pose a challenge, since they are sophisticated and  neither well documented nor easily readable.

mxcross commented:
@olaf-otto thank you for fixing the integration tests on main. I agree with your assessment that it will be challenging. At least now there's something that can be run through a debugger that works. Much appreciated!

mxcross committed:
Merge branch 'eclipse-gemini:main' into main

mxcross committed:
use java 17 image for upgraded branch

wilx commented:
Ping. How far is this project?

mxcross commented:
The lift to get the integration tests running/passing is greater than the upgrade itself.

AxeTheAxe commented:
Hi @mxcross ! What was the reason to close this PR without merging? The problems with the IT?

mxcross commented:
@AxeTheAxe Yup, the IT's weren't even in a compiling state and they are extremely complex. The lift to get those working again seems larger than the upgrade itself.

garpinc commented:
So which branch is the one we should be using? the eclipse-gemini:feature/582864-upgrade-to-spring-6 branch or this one? I Looks like this is the only one where all the PRs are merged in right? I'm on spring 6.1.6.1. 

Will I be able to get a build that works even though the IT tests don't? Are there binary jars somewhere so I don't have to fight through the compilation?

mxcross commented:
> So which branch is the one we should be using? the eclipse-gemini:feature/582864-upgrade-to-spring-6 branch or this one? I Looks like this is the only one where all the PRs are merged in right? I'm on spring 6.1.6.1.
> 
> Will I be able to get a build that works even though the IT tests don't? Are there binary jars somewhere so I don't have to fight through the compilation?

As long as you aren't using the "it" profile it should compile for you fine. It compiles, unit tests pass. I did have it functional locally.

arun-madan-1 commented:
@mxcross I noticed a new version of Gemini Blueprint (4.0.0-atlassian-4) available on maven central which supports spring 6. Do you have any idea if its good to use or not?
